v20.1/temporary-tables.md

```diff
@@ -0,0 +1,211 @@
+{{site.data.alerts.end}}
+
+## Details
+
+- Temp tables are automatically dropped at the end of the session.
+- A temp table can only be accessed from the session in which it was created.
+- Temp tables persist across transactions in the same session.
+- Persistent tables cannot reference temp tables, but temp tables can reference persistent tables.
+- Temp tables cannot be converted to persistent tables.
+
+## Temporary schemas
+
+Temp tables are not part of the `public` schema. Instead, when you create the first temp table for a session, CockroachDB generates a single temporary schema (`pg_temp_<id>`) for all of the temp tables, [temporary views](views.html#temporary-views), and [temporary sequences](create-sequence.html#temporary-sequences) in the current session for a database. In a session, you can reference the session's temporary schema as `pg_temp`.
+
+## Syntax
+
+To create a temp table, add the following [`TEMP`/`TEMPORARY`](sql-grammar.html#opt_temp_create_table) to the beginning of a [`CREATE TABLE`](create-table.html) or [`CREATE TABLE AS`](create-table-as.html) statement:
+
+<div>
+  {% include {{ page.version.version }}/sql/diagrams/opt_temp_create_table.html %}
+</div>
+
+
+## Example
+
+To turn on temp tables:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SET experimental_enable_temp_tables=on;
+~~~
+
+To create a temp table:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TEMP TABLE users (
+        id UUID PRIMARY KEY,
+        city STRING,
+        name STRING,
+        address STRING
+);
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW CREATE TABLE users;
+~~~
+
+~~~
+  table_name |                create_statement
+-------------+-------------------------------------------------
+  users      | CREATE TEMP TABLE users (
+             |     id UUID NOT NULL,
+             |     city STRING NULL,
+             |     name STRING NULL,
+             |     address STRING NULL,
+             |     CONSTRAINT "primary" PRIMARY KEY (id ASC),
+             |     FAMILY "primary" (id, city, name, address)
+             | )
+(1 row)
+~~~
+
+To show the newly created `pg_temp` schema, use [`SHOW SCHEMAS`](show-schemas.html):
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW SCHEMAS;
+~~~
+
+~~~
+           schema_name
+---------------------------------
+  crdb_internal
+  information_schema
+  pg_catalog
+  pg_temp_1584540651942455000_1
+  public
+~~~
+
+
+To create another temp table that references `users`:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TEMP TABLE vehicles (
+        id UUID NOT NULL,
+        city STRING NOT NULL,
+        type STRING,
+        owner_id UUID,
+        creation_time TIMESTAMP,
+        CONSTRAINT "primary" PRIMARY KEY (city ASC, id ASC),
+        INDEX vehicles_auto_index_fk_city_ref_users (city ASC, owner_id ASC)
+);
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW CREATE TABLE vehicles;
```

Comment:
Perhaps the SHOW CREATE is to prove that the index is pointing at the other table?